packages/types/src/types.ts

```diff
@@ -875,3 +875,46 @@ export const Path = typedef(
   },
   v => v
 )
+
+/**
+ * InclusiveRange type, step size defaults to 1
+ *
+ * @param t - A TypeDescriptor for the type of the range, must be a number (UInt32, Int32, etc.)
+ * @returns A TypeDescriptor for an InclusiveRange of the given type
+ *
+ * @example
+ * ```javascript
+ * import * as fcl from "@onflow/fcl"
+ * import {InclusiveRange, UInt32} from "@onflow/types"
+ *
+ * // Range of 1, 2, 3
+ * const test = fcl.arg({start: 1, end: 3}, InclusiveRange(UInt32))
+ *
+ * // Range of 1, 3, 5
+ * const test = fcl.arg({start: 1, end: 5, step: 2}, InclusiveRange(UInt32))
+ * ```
+ */
+export const InclusiveRange = <T extends TypeDescriptor<any, any>>(t: T) =>
+  typedef(
+    "InclusiveRange",
+    (v: {
+      start: TypeDescriptorInput<T>
+      end: TypeDescriptorInput<T>
+      step?: TypeDescriptorInput<T>
+    }) => {
```

Comment:
I would have prefered to use a tuple here for brevity, but it feels more explicit to use an object for the decoding, and I think there's value in encoding/decoding formats being in sync